File: Knight.java
package the_player;
public class Knight implements Player{
    public Knight(int s[]){
        stats = s;
    }
    @Override
    public void heal(int amount){
        stats[0] += amount;
        if (stats[0] > stats[1]){
            stats[0] = stats[1];
        }
    }
    @Override
    public void damage(int amount){
        amount -= 2;
        stats[0] -= amount;
        if (stats[0] <= 0){
            alive = false;
        }
    }
    @Override
    public void next_turn(){
        if (effects[1] > 0){
            effects[1]--;
            if (effects[1] == 0){
                effects[0] = 0;
            }
        }
    }
}

File: include/Knight.cpp
#include <iostream>
#include <vector>

#include "classes.hpp"

using namespace std;

Knight::Knight(int s[10], short i[5], string p, string a): Player(s, i, p, a){}
void Knight::damage(int amount){
    amount -= 2;
    stats[0] -= amount;
    dead = (stats[CURRENT_HP] <= 0);
}
void Knight::heal(int amount){
    stats[0] += amount;
    stats[0] = (stats[CURRENT_HP] > stats[MAX_HP]) ? stats[MAX_HP] : stats[CURRENT_HP];
}
void Knight::next_turn(){
    if (effects.adtr > 0){
        effects.adtr--;
        effects.adv = (effects.adtr == 0) ? 0 : effects.adv;
    }
    stats[CURRENT_MP] += stats[MP_REFRESH];
    stats[CURRENT_MP] = (stats[CURRENT_MP] > stats[MAX_MP]) ? stats[MAX_MP] : stats[CURRENT_MP];
}